### ISSUE TITLE
Wrong logic generate HTML dataset attribute

### ISSUE THREAD

**Issue by s1lver** [yiisoft/yii2#17868]
### What steps will reproduce the problem?

Previously, the generation of dataset attributes when passing true or false, converted to 1 or 0. Now the logic has changed. Maybe worth a review #17625?

### What is the expected result?

```php 
echo Html::tag('div', 'TEST', ['data' => ['id' => 'main_div', 'test' => true]]);
```

generate HTML code 

```html
<div data-id="main_div" data-test="1">TEST</div>
```

### What do you get instead?
**TRUE**

```php 
echo Html::tag('div', 'TEST', ['data' => ['id' => 'main_div', 'test' => true]]);
```

generate HTML code 

```html
<div data-id="main_div" data-test="">TEST</div>
```
**FALSE**

```php 
echo Html::tag('div', 'TEST', ['data' => ['id' => 'main_div', 'test' => false]]);
```

generate HTML code 

```html
<div data-id="main_div">TEST</div>
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.32
| PHP version      | 7.3.11
| Operating system | Unix

**Comment by s1lver** [comment]
I fix it
```php
elseif (is_bool($v)) {
    if ($v) {
        $html .= " $name-$n=" .$v;
    }
 } 
```

**Comment by alex-code** [comment]
Do you need the 1/0 on the attribute?
You could pass an int/string to get a value.

**Comment by s1lver** [comment]
The new logic provides for the presence of the attribute itself. It breaks backward compatibility

**Comment by alex-code** [comment]
I'd say the update brings it into line with the html spec, https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute

**Comment by samdark** [comment]
@s1lver even if it does, it was released in 2.0.29. It's three versions ago so already adopted. Revering it would be backwards compatibility break as well. What we can do is adding a note to UPGRADE.

**Comment by s1lver** [comment]
Ok
